src/scanner.rs

```diff
@@ -7,6 +10,22 @@ use sysinfo::{CpuExt, System, SystemExt};
 
 use crate::logo::*;
 
+#[derive(Serialize, Deserialize, Debug)]
+struct SPDisplay {
+    sppci_model: String,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+struct SPDisplays {
+    #[serde(rename = "SPDisplaysDataType")]
+    sp_displays_data_type: Vec<SPDisplay>,
+}
+
+lazy_static! {
+    static ref GPU_RE: Regex =
+        Regex::new(r#"^\S+? "(VGA|3D|Display).*?" ".*?" "(?P<gpu>.*?)""#).unwrap();
```

Comment:
I'm thinking `unwrap()` here might not be the best option - even though it does seem unlikely to fail since the regex has been tested already. If it does, however, I feel omitting routines dependent on `GPU_RE` could handled.

Reply:
Generally, I too am averse to `unwrap()` - [the docs discourage its use](https://doc.rust-lang.org/std/result/enum.Result.html#method.unwrap), although most people in the rust comminuty don't seem to heed this.

However, in this case, the error will only occur if the regex: `r#"^\S+? "(VGA|3D|Display).*?" ".*?" "(?P<gpu>.*?)""#` is not well defined, not if there is an error when mathing any particular string. Because the regex is a constant string, the value of `Regex::new(...)` will either always be an `Err`, or always be an `Ok`. There is no variation in runtime behaviour. So pushing the error handling down to the consumers of GPU_RE is a waste. (Incidenally, the fact that the regex is constant is also why I'm using the `lazy_static` macro to ensure it's only compiled once.)

Ideally, this is something that could be detected at compile time, but it might be outside the design of the compiler to check this. I think this is the next best thing: any attempt to use this module will break, even if no methods from the module are invoked. One way to make this more robust is to have a test that asserts that the regex does not panic. I suspect any test that has `use crate::scanner` in it will achieve this. I think I might verify this and add one.

Lmk what you think.